### PR TITLE
Flexible survey

### PR DESCRIPTION
**Pull request by amyrebecca**
This adds a new survey-related task to the system, "flexibleSurvey." FlexibleSurvey works like a regular survey, but adds the ability to specify which questions belong to which choices by the way that an additional field is filled out in the questions csv file.

It's of course silly to have two survey tasks, but it was considered too risky to modify the existing task directly because we have deployments who are consuming it.

I'm looking to merge this onto master tomorrow so that we can deploy it for WiscWildWatch

[sample spreadsheet](https://github.com/zooniverse/Panoptes-Front-End/files/69062/surveyTask-spreadsheets-inclHumanNnothing-multiQuestions-v2.xlsx)

### CODE REVIEW

**Comment by amyrebecca** [comment]
Brian, I know you're busy, but would it be possible for you to look at this? Everyone suggested you because you'd written the original survey task!

File: app/classifier/tasks/flexible-survey/annotation-view.cjsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    task: null
+    classification: null
+    annotation: null
+
+  render: ->
+    <div>
+      {for identification, i in @props.annotation.value
+        identification._key ?= Math.random()
+
+        #TODO: ARB: what is happening here
```
Comment (camallen) [inline app/classifier/tasks/flexible-survey/annotation-view.cjsx]: stray comment?

File: app/classifier/tasks/flexible-survey/choice.cjsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+          for index in [0...@props.images.length]
+            <span key={@props.images[index]}>
+              <button type="button" className="survey-task-image-flipper-pip" disabled={index is @state.frame} onClick={@handleFrameChange.bind this, index}>{index + 1}</button>
+              {' '}
+            </span>}
+      </span>
+    </span>
+
+  renderPreload: ->
+    <div style={@PRELOAD_STYLE}>
```
Comment (camallen) [inline app/classifier/tasks/flexible-survey/choice.cjsx]: should we be using css here instead?

**Comment by camallen** [comment]
This is a pretty big chunk of work, i won't have time to test it out properly sorry.

File: app/classifier/tasks/flexible-survey/utility.coffee [hunk 3]
```diff
@@ -0,0 +1,8 @@
+module.exports =
+  getQuestionIDs: (task, choiceID) ->
+    return [] if task.choices[choiceID].noQuestions
+    return task.questionsOrder unless task.questionsMap? and choiceID of task.questionsMap
+    task.questionsMap[choiceID]
+  getQuestions: (task, choiceID) ->
+    @getQuestionIDs(task, choiceID).map (idx ->
```
Comment (aweiksnar) [inline app/classifier/tasks/flexible-survey/utility.coffee]: I think this function is unused, but if I'm wrong `idx` would need to be wrapped in parens

File: app/classifier/tasks/flexible-survey/choice.cjsx (hunk 2 above)
Comment (brian-c) [inline app/classifier/tasks/flexible-survey/choice.cjsx]: This style is required for the module to function right, so I think including it in the module is the right call.

**Comment by aweiksnar** [comment]
:cool: :cool: Also for github history's sake: running `diff app/classifier/tasks/survey/ app/classifier/tasks/flexible-survey/` gives a better representation of the files changed from the original survey task

**Pull request merged**